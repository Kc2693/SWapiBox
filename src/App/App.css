@import url(https://fonts.googleapis.com/css?family=Audiowide);

:root {
    --main-yellow: #E58E58;
    --main-light-red: #9d323b;
    --main-dark-red: #6E233A;
    --button-text-color: white;
}
.App-container {
  height: 100vh;
  width: 100%;
}

.Background {
  background: url(https://orig00.deviantart.net/9fb6/f/2015/334/2/7/shooting_stars2_by_kirokaze-d9ikcm1.gif) no-repeat;
  background-position: 50% 0;
  position: absolute;
  height: 100vh;
  width: 100%;
  z-index: -10;
}
.Background:after {
  content: " ";
  display: block;
  position: absolute;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.4);
}

.Foreground {
  height: 100%;
  position: relative;
  z-index: 1;
}
.main-title {
  color: var(--main-yellow);
  font-family: 'Audiowide', cursive;
  font-size: 50px;
  margin: auto;
  width: 280px;
  z-index: 10;
}
.buttons-container {
  display: flex;
  justify-content: space-around;
  width: 500px;
  margin: 50px auto;
}
