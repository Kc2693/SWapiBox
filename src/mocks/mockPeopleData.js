module.exports = [
{"name":"Luke Skywalker","homeworld":"Tatooine","population":"200000","species":"Human","language":"Galactic Basic"},
{"name":"C-3PO","homeworld":"Tatooine","population":"200000","species":"Droid","language":"n/a"},
{"name":"R2-D2",
"homeworld":"Naboo",
"population":"4500000000",
"species":"Droid",
"language":"n/a"},
{"name":"Darth Vader",
"homeworld":"Tatooine",
"population":"200000",
"species":"Human",
"language":"Galactic Basic"},
{"name":"Leia Organa",
"homeworld":"Alderaan",
"population":"2000000000",
"species":"Human",
"language":"Galactic Basic"}
]
