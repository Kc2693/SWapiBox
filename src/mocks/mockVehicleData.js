module.exports = [
		{
			"name": "Sand Crawler",
			"model": "Digger Crawler",
			"passengers": "30",
			"vehicle_class": "wheeled",
		},
		{
			"name": "T-16 skyhopper",
			"model": "T-16 skyhopper",
			"passengers": "1",
			"vehicle_class": "repulsorcraft",
		},
		{
			"name": "X-34 landspeeder",
			"model": "X-34 landspeeder",
			"passengers": "1",
			"vehicle_class": "repulsorcraft",
		},
		{
			"name": "TIE/LN starfighter",
			"model": "Twin Ion Engine/Ln Starfighter",
			"passengers": "0",
			"vehicle_class": "starfighter",
		}
]
