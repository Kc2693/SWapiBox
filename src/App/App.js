import React, { Component } from 'react';
import './App.css';
import Sidebar from '../Sidebar/Sidebar';
import Button from '../Button/Button';
class App extends Component {










  render() {
    return (
      <div className="App-container">
      <div className="Background"></div>
        <div className="Foreground">
          <h1 className="main-title">SWapiBox</h1>
          <div className="buttons-container">
          <Button title={'people'}/>
          <Button title={'planets'}/>
          <Button title={'vehicles'}/>
          </div>


          <Sidebar />

      </div>
      </div>
    );
  }
}

export default App;
