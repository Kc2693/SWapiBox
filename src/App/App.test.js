import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import LocalStorage from '../mocks/mockLocalStorage';
import mockVehicleData from '../mocks/mockVehicleData';
import mockPeopleData from '../mocks/mockPeopleData'

describe('App', () => {
  let wrapper;
  let mockFetchFilmTexts = jest.fn().mockImplementation(() => {
        return PmockScrollText
      });
  let mockScrollText;
  let fetchPeopleInfo;

  beforeEach(() => {
    window.localStorage = new LocalStorage();
    wrapper = shallow(<App fetchFilmTexts={mockFetchFilmTexts}/>, {disableLifecycleMethods: true})
    mockScrollText = [
      {
      "title": "A New Hope",
      "opening_crawl": "It is a period of civil war.",
      "release_date": "1977-05-25"
    }, {
      "title": "A New Hope",
      "opening_crawl": "It is a period of civil war.",
      "release_date": "1977-05-25"
    }, {
      "title": "A New Hope",
      "opening_crawl": "It is a period of civil war.",
      "release_date": "1977-05-25"
    }]

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve({ })
    }))
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should fetch a random film text upon componentDidMount', () => {

    expect(wrapper.state('scrollText')).toEqual(mockScrollText)
  });

  it('should set category state to true if corresponding button is clicked', () => {
    const expected = {'people': true, 'planets': false, 'vehicles': false}

    wrapper.instance().selectCategory('people');
    expect(wrapper.state('categories')).toEqual(expected)
  });

  it('should set all state categories to false if corresponding button is clicked twice', () => {
    const afterFirstClick = {'people': true, 'planets': false, 'vehicles': false}
    const afterSecondClick = {'people': false, 'planets': false, 'vehicles': false}

    wrapper.instance().selectCategory('people');
    expect(wrapper.state('categories')).toEqual(afterFirstClick);

    wrapper.instance().selectCategory('people');
    expect(wrapper.state('categories')).toEqual(afterSecondClick);
  });

  it('should display info container if a category is true', () => {
    wrapper.instance().selectCategory('people');
    wrapper.instance().displayInfoContainer();

    expect(wrapper.state('categories').people).toBe(true);
    expect(wrapper.find('InfoContainer')).toBeDefined();
  });

  it.skip('fetchCategory should take in a button name and fetch the corresponding category from storage', async () => {
    await localStorage.setItem('vehicles', JSON.stringify(mockVehicleData));

    expect(localStorage.getItem('vehicles')).toEqual(JSON.stringify(mockVehicleData))
 // this test is not getting past if (!localStorage[key]) for some reason?
    await wrapper.instance().fetchCategory('vehicles');
  });

  it.skip('fetchCategory should make a fetch call if there is no corresponding category in localstorage', () => {
    // this is not using my mock fetch and is instead trying to do a real fetch
    fetchPeopleInfo = jest.fn().mockImplementation(() => {
      return Promise.resolve(mockPeopleData)
    })
    expect(localStorage.getItem('people')).toEqual(undefined);

    wrapper.instance().fetchCategory('people');

    expect(localStorage.getItem('people')).toEqual(JSON.stringify(mockPeopleData));
    expect(wrapper.state('people')).toEqual(mockPeopleData);
  });
});
