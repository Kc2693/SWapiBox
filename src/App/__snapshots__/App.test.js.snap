// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchFilmTexts={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Background"
/>,
        <div
          className="Foreground"
>
          <h1
                    className="main-title"
          >
                    SWapiBox
          </h1>
          <Button
                    controlFunc={[Function]}
                    faveNum={0}
                    id="faves"
                    name="Favorites "
          />
          <categoryButtonContainer
                    categories={
                              Array [
                                        "people",
                                        "planets",
                                        "vehicles",
                                      ]
                    }
                    controlFunc={[Function]}
                    fetchFunc={[Function]}
          />
</div>,
      ],
      "className": "App-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="main-title"
>
              SWapiBox
</h1>,
            <Button
              controlFunc={[Function]}
              faveNum={0}
              id="faves"
              name="Favorites "
/>,
            <categoryButtonContainer
              categories={
                            Array [
                                          "people",
                                          "planets",
                                          "vehicles",
                                        ]
              }
              controlFunc={[Function]}
              fetchFunc={[Function]}
/>,
            false,
            Array [],
          ],
          "className": "Foreground",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SWapiBox",
              "className": "main-title",
            },
            "ref": null,
            "rendered": "SWapiBox",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "controlFunc": [Function],
              "faveNum": 0,
              "id": "faves",
              "name": "Favorites ",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "categories": Array [
                "people",
                "planets",
                "vehicles",
              ],
              "controlFunc": [Function],
              "fetchFunc": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Background"
/>,
          <div
            className="Foreground"
>
            <h1
                        className="main-title"
            >
                        SWapiBox
            </h1>
            <Button
                        controlFunc={[Function]}
                        faveNum={0}
                        id="faves"
                        name="Favorites "
            />
            <categoryButtonContainer
                        categories={
                                    Array [
                                                "people",
                                                "planets",
                                                "vehicles",
                                              ]
                        }
                        controlFunc={[Function]}
                        fetchFunc={[Function]}
            />
</div>,
        ],
        "className": "App-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="main-title"
>
                SWapiBox
</h1>,
              <Button
                controlFunc={[Function]}
                faveNum={0}
                id="faves"
                name="Favorites "
/>,
              <categoryButtonContainer
                categories={
                                Array [
                                                "people",
                                                "planets",
                                                "vehicles",
                                              ]
                }
                controlFunc={[Function]}
                fetchFunc={[Function]}
/>,
              false,
              Array [],
            ],
            "className": "Foreground",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SWapiBox",
                "className": "main-title",
              },
              "ref": null,
              "rendered": "SWapiBox",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "controlFunc": [Function],
                "faveNum": 0,
                "id": "faves",
                "name": "Favorites ",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "categories": Array [
                  "people",
                  "planets",
                  "vehicles",
                ],
                "controlFunc": [Function],
                "fetchFunc": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
